resolwe/observers/protocol.py

```diff
@@ -22,3 +23,26 @@ class ChangeType(Enum):
 # or before and after a PermissionObject's container is changed.
 pre_permission_changed = dispatch.Signal(providing_args=["instance"])
 post_permission_changed = dispatch.Signal(providing_args=["instance"])
+
+# Signal to be sent before and after container is changed.
+pre_container_changed = dispatch.Signal(providing_args=["instance"])
+post_container_changed = dispatch.Signal(providing_args=["instance"])
+
+
+class ChannelsMessage(TypedDict):
+    """The type for channels message to be sent."""
+
+    type: str
+    object_id: Optional[int]
+    content_type_pk: int
+    change_type_value: int
+    source: Optional[tuple[str, int]]
```

Comment:
`object_id` and `content_type_pk` are redundant, since they're already in `source`

Reply:
See above.